rasr/command.py

```diff
@@ -132,17 +132,29 @@ def run_script(
         args: List = None,
         retries: int = 2,
         use_tmp_dir: bool = False,
+        copy_tmp_ls: Optional[List] = None,
     ):
         args = [] if args is None else args
         tmp_log_file = remove_suffix(
             os.path.basename(tk.uncached_path(log_file)), ".gz"
         )
 
+        work_dir = os.path.abspath(os.curdir)
         if use_tmp_dir:
             date_time_cur = time.strftime("%y%m%d-%H%M%S", time.localtime())
             with tempfile.TemporaryDirectory(prefix=gs.TMP_PREFIX) as tmp_dir:
                 print("using temp-dir: %s" % tmp_dir)
                 try:
+                    if copy_tmp_ls is None:
+                        assert (
+                            task_id == 1
+                        ), "Concurrent Jobs need a list of files to copy due to race conditions"
+                        file_names = os.listdir(work_dir)
+                    else:
+                        file_names = copy_tmp_ls
+                    for fn in file_names:
+                        print(fn)
+                        shutil.copy(os.path.join(work_dir, fn), "%s/%s" % (tmp_dir, fn))
                     self.run_cmd(cmd, [task_id, tmp_log_file] + args, retries, tmp_dir)
                     file_names = os.listdir(tmp_dir)
                     for fn in file_names:
```

Comment:
do we need a similar logic also here? E.g. only copy files back that are _not_ in the `copy_tmp_ls`.